istioctl/cmd/workload.go

```diff
@@ -388,11 +388,15 @@ func createMeshConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.Workloa
 func createHosts(kubeClient kube.ExtendedClient, ingressIP, dir string) error {
 	// try to infer the ingress IP if the provided one is invalid
 	if validation.ValidateIPAddress(ingressIP) != nil {
-		ingress, err := kubeClient.CoreV1().Services(istioNamespace).Get(context.Background(), multicluster.IstioIngressGatewayServiceName, metav1.GetOptions{})
-		if err == nil && ingress.Status.LoadBalancer.Ingress != nil && len(ingress.Status.LoadBalancer.Ingress) > 0 {
-			ingressIP = ingress.Status.LoadBalancer.Ingress[0].IP
+		ingress, err := kubeClient.CoreV1().Services(istioNamespace).Get(context.Background(), multicluster.IstioEastWestGatewayServiceName, metav1.GetOptions{})
```

Comment:
since eastwest gateway is just a convention not a requirement should we make this configurable? We will want to support things like external istiod for example, or just a custom gateway setup

Reply:
Good point I'll add `discoveryServiceName` or something similar

Reply:
We already allow specifying the IP directly, but service name might be better UX. We do need to give some kind of output here if the IP isn't available (still pending for example).

Reply:
does dns name also work or just direct ip? since that will likely be used for external.

we could also take in a hostname instead of service and resolve to service ip if it's internal possibly, although not sure how challenging that is. lot of options

Reply:
Right now this is used to create a /etc/hosts "addendum", then in the proxy config we have discovery address as `istiod.istio-system...` - I figure the DNS name would have to be used as discovery address. 

I think tying the gateway to a k8s service is the common case. if they're doing something special we should cover the DNS name case, the manual IP case

As far as resolving a hostname.. not sure how we'd do that? Are we trying to use the address that we resolve to and use the ClusterIP? I figure looking up by name is more reliable. I can see a `<svc>.<namespace>` like syntax being good so we're not restricted to the system namespace.

Reply:
Is this good for now? I'd like to be able to remove the "get up" step from the docs.